src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.Types.cs

```diff
@@ -0,0 +1,300 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Xunit;
+
+public static class StaticClass
```

Comment:
Should we also add a test with some unspeakable type that has characters that might trip the parser over? Eg. in our sidecar .dll we might have a type like this:
```
ABI.Windows.Foundation.<#Windows>TypedEventHandler`2<<#Windows>Windows-ApplicationModel-Activation-SplashScreen|object>Impl
```

I wonder if that backtick character might cause issues with the parsing logic that expects to have a generic type (whereas this type is non-generic, it just has the backtick character in its name). I'm not entirely sure `Type.GetType` properly handles it either, but perhaps even just validating that the two are consistent could make sense? 🤔

Reply:
The backtick is entirely a "source language to IL compiler" convention. Nothing in ECMA requires that convention, but following it provides a mechanism for languages to support providing multiple classes with the same name with different generic parameter counts in source.

Reply:
> Should we also add a test with some unspeakable type that has characters that might trip the parser over?

We follow section I.10.1 in ECMA-335 for this. If a system produces names that are inconsistent (for example, inconsistent Unicode combining) then that is a bug in the other system.